cdc/sink/producer/kafka/kafka.go

```diff
@@ -395,80 +409,120 @@ func (k *kafkaSaramaProducer) run(ctx context.Context) error {
 	}
 }
 
-// kafkaTopicPreProcess gets partition number from existing topic, if topic doesn't
-// exit, creates it automatically.
-func kafkaTopicPreProcess(topic, address string, config *Config, cfg *sarama.Config) (int32, error) {
-	admin, err := sarama.NewClusterAdmin(strings.Split(address, ","), cfg)
+func topicPreProcess(config *Config, saramaConfig *sarama.Config) error {
+	failpoint.Inject("workaround4Test", func() {
```

Comment:
Could you avoid failpoints in unit tests? Can it be replaced by mocking `topicPreProcess`?

Reply:
It was quite hard to mock this method, since this method is defined in the package `sink/kafka`, but will also be called by unit test in the `sink` package.

If try to mock the method, have to make this method public, I think it's unreasonable.

failpoint would be the easiest way to workaround this.

Reply:
Maybe we can leave a `FIXME` here for now?